### DISCUSSION TITLE
How to handle nulls when scanning by name .RowsToStructByName[Lax]

### DISCUSSION THREAD

**Discussion by sai-sy** [jackc/pgx#1972] [Q&A]
I'm trying to get a row scanned into my struct, and I got most of the way through, but one of my rows is missing some data, and I'm wondering how to handle that since when using
```go
r, err := pgx.CollectRows(rows, pgx.RowToStructByNameLax[person])
```
I get the error
```bash
2024/03/28 02:14:33 can't scan into dest[5]: cannot scan NULL into *time.Time
```
and the documentation doesn't seem the mention anything (unless I am just reading right over it

Also does PGX support casting into strings on scan. I had to import time to handle `created_by` which is fine but postgres has some deeper types that might get weird to cast into no? That is on the same note as handling null but not entirely so not sure if it warrants it's own discussion

**Comment by sai-sy** [comment]
Okay, I see that I can set the field type to an interface and then check case for each one. That seems like a poor pattern, would there be a different way to handle this?

**Comment by jackc** [comment]
The problem is not specific to `pgx.RowToStructByNameLax`. The problem is `time.Time` cannot represent `NULL`. You would get the same issue if you were calling `Scan` yourself.

The solution is to use a type that can represent a `NULL` such as `*time.Time`, `pgtype.Timestamptz`, or `zeronull.Timestamptz`.
